src/components/Carousel.tsx

```diff
@@ -30,46 +38,68 @@ function Carousel(props: CarouselProps): JSX.Element {
+    props.onPrev && props.onPrev();
+  }
+
+  const child = props.children[slideIdx];
+  if (child === undefined && props.children.length > 0) {
+    const slideInBounds = Math.min(Math.max(0, slideIdx), props.children.length - 1);
+    setSlideIdx(slideInBounds);
+  }
+
   return (
-    <div id={'carousel-wrapper'}>
-      { props.title && <h1 id={'title'}>{props.title}</h1> }
-      { props.subtitle && <h2 id={'subtitle'}>{props.subtitle}</h2> }
-      <div id={'carousel'}>
-        <button
-          className={'carousel-btn prev'}
-          style={{
-            visibility: ((props.showPrev !== undefined && props.showPrev) || (slideIdx > 0))
-              ? 'visible'
-              : 'hidden',
-          }}
-          onClick={() => {
-            setSlideIdx(old => Math.max(old - 1, 0));
-            props.onPrev && props.onPrev();
-          }}
-        >
-          <img src={PrevSvg} />
-        </button>
-        <div id={'carousel-content'}>
-          {props.children && props.children.length > 0 && props.children[slideIdx]}
-        </div>
-        <button
-          className={'carousel-btn next'}
-          style = {{
-            visibility:
-              ((props.showNext !== undefined && props.showNext) ||
-               (slideIdx < (props.children ? props.children.length - 1 : 0)))
+    <ChangeSlideContext.Provider value={{next: goNext, prev: goPrev}}>
+      <div id={'carousel-wrapper'}>
+        {props.title && <h1 id={'title'}>{props.title}</h1>}
+        {props.subtitle && <h2 id={'subtitle'}>{props.subtitle}</h2>}
+        <div id={'carousel'}>
+          <button
+            className={'carousel-btn prev'}
+            style={{
+              visibility: child &&
+                ((child.showPrev === undefined && slideIdx > 0) ||
+                  (child.showPrev === true))
                 ? 'visible'
                 : 'hidden',
-          }}
-          onClick={() => {
-            setSlideIdx(old => Math.min(old + 1, props.children ? props.children.length - 1 : 0));
-            props.onNext && props.onNext();
-          }}
-        >
-          <img src={NextSvg} />
-        </button>
+            }}
+            onClick={() => goPrev()}
+          >
+            <img src={PrevSvg} />
+          </button>
+          <div id={'carousel-content'}>
+            {
+              child &&
+              <div id={'carousel-items'}>
+                {child.topText && <h2 id={'body-text'}> {child.topText} </h2>}
+                {child.child}
+                {child.bottomText && <h2 id={'body-text'}> {child.bottomText} </h2>}
+              </div>
+            }
+          </div>
+          <button
+            className={'carousel-btn next'}
+            style={{
+              visibility:
+                (child &&
+                  (child.showNext === undefined && (slideIdx < props.children.length - 1))
+                  || (child.showNext === true))
+                  ? 'visible'
+                  : 'hidden',
```

Comment:
same as above

nit:
```suggestion
              visibility: child?.showNext === false || slideIdx === props.children.length - 1
                  ? 'hidden'
                  : 'visible,
```